### PR TITLE
`Modeling and Text Exercises`: View example solutions

### PR DESCRIPTION
**Pull request by terlan98**
> ⚠️ Should be merged after #170 

This PR allows viewing example solutions for modeling and text exercises. The example solution is shown in CorrectionSidebar, below the problem statement.

### CODE REVIEW

File: Themis/ViewModels/Assessment/Modeling/UMLRendererViewModel.swift [hunk 1]
```diff
@@ -113,6 +137,18 @@ class UMLRendererViewModel: ExerciseRendererViewModel {
...
     }
     
+    @MainActor
+    /// Sets the drag location to the specified point
+    /// - Parameter point: when nil, the drag location is set in such a way that centers the diagram
+    func setDragLocation(at point: CGPoint? = nil) {
+        if let point {
+            currentDragLocation = point
+        } else {
+            currentDragLocation = .init(x: diagramSize.height - 50, // 50 – default padding added by Athena
```
Comment (maximiliansoelch) [inline Themis/ViewModels/Assessment/Modeling/UMLRendererViewModel.swift]: What does the padding for modeling exercises have to do with Athena?

File: Themis/Views/Assessment/CorrectionSidebar/CorrectionSidebarView.swift [hunk 2]
```diff
@@ -35,16 +35,19 @@ struct CorrectionSidebarView: View {
...
+                ZStack {
+                    ScrollView {
+                        problemStatement
+                        exampleSolution
+                    }
+                    .padding(.horizontal, problemStatementNeedsPadding ? 15 : 0)
+                    .opacity(correctionSidebarStatus == .problemStatement ? 1.0 : 0.0001)
+                    // 0.0 causes this view to be redrawn and webview to send a new request
+                    
+                    if !assessmentVM.loading {
```
Comment (maximiliansoelch) [inline Themis/Views/Assessment/CorrectionSidebar/CorrectionSidebarView.swift]: Is the second loading check required? 🤔

**Review by maximiliansoelch — CHANGES_REQUESTED** [review]
Is this intended?

https://github.com/ls1intum/Themis/assets/9954674/ae8f0e13-c6c9-4c09-ba00-0701bce42650

Also, the example solution does not support skeleton loading.

File: Themis/ViewModels/Assessment/Modeling/UMLRendererViewModel.swift (hunk 1 above)
Comment (terlan98) [reply Themis/ViewModels/Assessment/Modeling/UMLRendererViewModel.swift]: Nothing, I wanted to write Apollon 😅

File: Themis/Views/Assessment/CorrectionSidebar/CorrectionSidebarView.swift (hunk 2 above)
Comment (terlan98) [reply Themis/Views/Assessment/CorrectionSidebar/CorrectionSidebarView.swift]: Actually, the first check is unnecessary. I removed that and kept this one.

**Comment by terlan98** [comment]
Thanks for the review, @maximiliansoelch 👍 
I addressed your comments. 

Regarding the video: I think it's not an issue because if the fingers are located on different canvases during the **pinch** gesture, it becomes two separate **drag** gestures, affecting both canvases. Here's a video showing this behavior in more detail:

https://github.com/ls1intum/Themis/assets/22798314/945b5706-4d3b-4f76-ac18-d767408cd035

**Review by maximiliansoelch — APPROVED** [review]

**Pull request merged**